lib/CardDAV/plugin.js

```diff
@@ -410,6 +410,18 @@ var jsCardDAV_Plugin = module.exports = jsDAV_Plugin.extend({
         if (vobj.name != "VCARD")
             throw new Exc.UnsupportedMediaType("This collection can only support vcard objects.");
 
+        // This fix an issue when syncing with MacOS addressbook: the UID of the object
+        // is not in vobj but in its first child
+        if (!vobj.UID) {
+            if (Array.isArray(vobj.children)) {
+                vobj.children.forEach(function (child) {
+                    if (child.name === 'UID')
```

Comment:
@clochix no `{` on this line, is it me or it should be added to match the `}` on line 420?